pytest_warnings.py

```diff
@@ -5,11 +5,44 @@
 import warnings
 
 
+def _setoption(wmod, arg):
+    """
+    Copy of the warning._setoption function but does not escape arguments.
+    """
+    parts = arg.split(':')
+    if len(parts) > 5:
+        raise wmod._OptionError("too many fields (max 5): %r" % (arg,))
+    while len(parts) < 5:
+        parts.append('')
+    action, message, category, module, lineno = [s.strip()
+                                                 for s in parts]
+    action = wmod._getaction(action)
+    category = wmod._getcategory(category)
+    if lineno:
+        try:
+            lineno = int(lineno)
+            if lineno < 0:
+                raise ValueError
+        except (ValueError, OverflowError):
+            raise wmod._OptionError("invalid lineno %r" % (lineno,))
+    else:
+        lineno = 0
+    wmod.filterwarnings(action, message, category, module, lineno)
+
+
+
 def pytest_addoption(parser):
     group = parser.getgroup("pytest-warnings")
     group.addoption(
         '-W', '--pythonwarnings', action='append',
-        help="set which warnings to report, see ...")
+        help="set which warnings to report, see -W option of python itself.")
+    group.addoption(
+        '--filterwarnings', action='append',
```

Comment:
Just a small comment:

We might not need the command-line option because since pytest-3.0 there's the new `-o` option which overrides ini-options so this is possible:

```
pytest "-o filterwarnings=ignore::UserWarning"
```

But I'm not sure how that plays when passed multiple times.